### PR TITLE
DAS-294: GUI and code for report generation

### PR DESCRIPTION
**Pull request by AndrewSazonov**

### CODE REVIEW

File: easyAppGui/Logic/Plotting.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+// ChemDoodle
+
+function chemDoodleInfo() {
+    return {
+        version: '9.1.0',
+        url: 'https://web.chemdoodle.com'
+    }
+}
+
```
Comment (rozyczko) [inline easyAppGui/Logic/Plotting.js:32]: ChemDoodle Web Components seem to be GPL-ed, so we need to include the license/acknowledgement for their use

File: easyAppGui/Logic/Plotting.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+// Bokeh
+
+function bokehInfo() {
+    return {
+        version: '2.2.3',
+        url: 'https://docs.bokeh.org/en'
+    }
+}
```
Comment (rozyczko) [inline easyAppGui/Logic/Plotting.js:108]: Bokeh uses BSD license and they also want it to be included.
https://github.com/bokeh/bokeh/blob/main/LICENSE.txt

**Review by rozyczko — APPROVED** [review]
very minor comments, the code looks good.

File: easyAppGui/Logic/Plotting.js (hunk 2 above)
Comment (AndrewSazonov) [reply easyAppGui/Logic/Plotting.js:108]: Done

File: easyAppGui/Logic/Plotting.js (hunk 1 above)
Comment (AndrewSazonov) [reply easyAppGui/Logic/Plotting.js:32]: Done

File: easyAppGui/Components/PreferencesDialog.qml [hunk 3]
```diff
@@ -26,46 +21,51 @@ EaElements.Dialog {
...
+                        EaStyle.Colors.theme = EaStyle.Colors.SystemTheme
+                }
+                Component.onCompleted: {
+                    ExGlobals.Variables.themeSelector = themeSelector
+                    if (EaStyle.Colors.theme === EaStyle.Colors.LightTheme)
+                        currentIndex = 0
+                    else if (EaStyle.Colors.theme === EaStyle.Colors.DarkTheme)
+                        currentIndex = 1
+                    else if (EaStyle.Colors.theme === EaStyle.Colors.SystemTheme)
+                        currentIndex = 2
```
[the remark this answers is not in the document]
Comment (AndrewSazonov) [reply easyAppGui/Components/PreferencesDialog.qml:58]: This is not to change the color scheme. It is to set (restore) the correct index in the drop-down list based on the saved color scheme value when you start the application.

**Pull request merged**